pkg/controller/direct/workstations/config_controller.go

```diff
@@ -0,0 +1,264 @@
+	// TODO: Support URLs
+	return nil, nil
+}
+
+type WorkstationConfigAdapter struct {
+	id        *krm.WorkstationConfigRef
+	gcpClient *gcp.Client
+	desired   *krm.WorkstationConfig
+	actual    *pb.WorkstationConfig
+}
+
+var _ directbase.Adapter = &WorkstationConfigAdapter{}
+
+func (a *WorkstationConfigAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting WorkstationConfig", "name", a.id.External)
+
+	req := &pb.GetWorkstationConfigRequest{Name: a.id.External}
+	workstationconfigpb, err := a.gcpClient.GetWorkstationConfig(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting WorkstationConfig %q: %w", a.id.External, err)
+	}
+
+	a.actual = workstationconfigpb
+	return true, nil
+}
+
+func (a *WorkstationConfigAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating WorkstationConfig", "name", a.id.External)
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+	resource := WorkstationConfigSpec_ToProto(mapCtx, &desired.Spec, nil)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	parent, err := a.id.Parent()
+	if err != nil {
+		return err
+	}
+	req := &pb.CreateWorkstationConfigRequest{
+		Parent:              parent.String(),
+		WorkstationConfigId: a.id.Name,
+		WorkstationConfig:   resource,
+	}
+	op, err := a.gcpClient.CreateWorkstationConfig(ctx, req)
+	if err != nil {
+		return fmt.Errorf("creating WorkstationConfig %s: %w", a.id.External, err)
+	}
+	created, err := op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("WorkstationConfig %s waiting creation: %w", a.id.External, err)
+	}
+	log.V(2).Info("successfully created WorkstationConfig", "name", a.id.External)
+
+	status := &krm.WorkstationConfigStatus{}
+	status.ObservedState = WorkstationConfigObservedState_FromProto(mapCtx, created)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	status.ExternalRef = &a.id.External
+	return createOp.UpdateStatus(ctx, status, nil)
+}
+
+func (a *WorkstationConfigAdapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("updating WorkstationConfig", "name", a.id.External)
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+	resource := WorkstationConfigSpec_ToProto(mapCtx, &desired.Spec, a.actual)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	// Set name and etag manually, because they are not filled-in by WorkstationConfigSpec_ToProto.
+	resource.Name = a.id.External
+	resource.Etag = a.actual.Etag
+
+	paths, err := common.CompareProtoMessage(resource, a.actual, common.BasicDiff)
+	if err != nil {
+		return err
+	}
+	if len(paths) == 0 {
+		log.V(2).Info("no field needs update", "name", a.id.External)
+		return nil
```

Comment:
Ah! Looks like the template is wrong. If this is acquiring a resource, the GCP resource may already in-sync with KCC  but KCC `status` is empty if not updated.

Reply:
Will you make this fix for the template? I will update the code in this controller to set status.